src/RelayClient.ts

```diff
@@ -0,0 +1,130 @@
+import type { RelayMetadata } from '@rsksmart/rif-relay-common';
+import {
+  EnvelopingTypes,
+  IForwarder__factory,
+} from '@rsksmart/rif-relay-contracts';
+import config from 'config';
+import { constants, ethers } from 'ethers';
+import AccountManager from './AccountManager';
+import type { HubInfo, RelayOrDeployRequest } from './common/relay.types';
+import EnvelopingEventEmitter from './events/EnvelopingEventEmitter';
+
+type RelayOnlyRequestKey = keyof Omit<
+  EnvelopingTypes.RelayRequestStruct['request'],
+  keyof EnvelopingTypes.DeployRequestStruct['request']
+>;
+const relayOnlyRequestKeys: RelayOnlyRequestKey[] = ['gas'];
+
+type DeployOnlyRequestKey = keyof Omit<
+  EnvelopingTypes.DeployRequestStruct['request'],
+  keyof EnvelopingTypes.RelayRequestStruct['request']
+>;
+const deployOnlyRequestKeys: DeployOnlyRequestKey[] = ['index', 'recoverer'];
+
+const isRelayRequest = (
+  request: EnvelopingTypes.RelayRequestStruct
+): request is EnvelopingTypes.RelayRequestStruct => {
+  return relayOnlyRequestKeys.every((key: RelayOnlyRequestKey) => {
+    return key in request;
+  });
+};
+
+const isDeployRequest = (
+  request: EnvelopingTypes.DeployRequestStruct
+): request is EnvelopingTypes.DeployRequestStruct => {
+  return deployOnlyRequestKeys.every((key: DeployOnlyRequestKey) => {
+    return key in request;
+  });
+};
+
+type RequestConfig = {
+  isSmartWalletDeploy?: boolean;
+  smartWalletAddress?: string;
+  clientId?: string;
```

Comment:
I dont see any use for the clientId

Reply:
just following the original implementation. Should I remove it, do you reckon?

Reply:
Ill wait for @antomor opinion.